detector/vuls2/vuls2.go

```diff
@@ -0,0 +1,650 @@
+			if len(ospkgs) <= index {
+				return nil, false, xerrors.Errorf("Too large OSPackage index. len(OSPackage): %d, index: %d", len(ospkgs), index)
+			}
+			statuses = append(statuses, models.PackageFixStatus{
+				Name: affectedPackageName(family, ospkgs[index]),
+				FixState: func() string {
+					if cn.Criterion.Version.FixStatus == nil {
+						return ""
+					}
+					return cn.Criterion.Version.FixStatus.Vendor
+				}(),
+				FixedIn:     fixedIn,
+				NotFixedYet: fixedIn == "",
+			})
+		}
+
+	}
+	return statuses, false, nil
+}
+
+func collectCVEs(e ecosystemTypes.Ecosystem, family string, dres detectTypes.DetectResult, am map[dataTypes.RootID]map[sourceTypes.SourceID]taggedAdvisory) map[dataTypes.RootID]map[vcTypes.VulnerabilityID]sourceVulnInfo {
+	m := make(map[dataTypes.RootID]map[vcTypes.VulnerabilityID]sourceVulnInfo)
+
+	for _, detected := range dres.Detected {
+		for _, dv := range detected.Vulnerabilities {
+			for sourceID, rm := range dv.Contents {
+				for rootID, vs := range rm {
+					for _, v := range vs {
+						if !includesEcosystem(v.Segments, e) {
+							continue
+						}
+						if _, found := m[rootID]; !found {
+							m[rootID] = make(map[vcTypes.VulnerabilityID]sourceVulnInfo)
+						}
+
+						a := func() *advisoryTypes.Advisory {
+							sourceIDMap, found := am[rootID]
+							if !found {
+								return nil
+							}
+							a, found := sourceIDMap[sourceID]
+							if !found {
+								return nil
+							}
+							return &a.advisory
+						}()
+						da := toDistoAdvisory(a)
+						if ignoresVulnerability(family, v, a) {
+							continue
+						}
+
+						cvss2, cvss3 := toCvss(v, da)
+
+						ccType := models.NewCveContentType(family)
+						cc := models.CveContents{
+							ccType: []models.CveContent{
+								{
+									Type:  ccType,
+									CveID: string(v.Content.ID),
+									Title: func() string {
+										if a != nil && a.Content.Title != "" {
+											return a.Content.Title
+										}
+										return v.Content.Title
+									}(),
+									Summary: func() string {
+										if a != nil && a.Content.Description != "" {
+											return a.Content.Description
+										}
+										return v.Content.Description
+									}(),
+									Cvss2Score:    cvss2.BaseScore,
+									Cvss2Vector:   cvss2.Vector,
+									Cvss2Severity: cvss2.NVDBaseSeverity,
+									Cvss3Score:    cvss3.BaseScore,
+									Cvss3Vector:   cvss3.Vector,
+									Cvss3Severity: cvss3.BaseSeverity,
+									SourceLink:    cveContentSourceLink(ccType, v),
+									References: func() models.References {
+										refs := v.Content.References
+										if a != nil {
+											refs = a.Content.References
+										}
+										rs := make([]models.Reference, 0, len(refs))
+										for _, r := range refs {
+											if a != nil && (strings.Contains(r.URL, string(a.Content.ID)) || strings.Contains(r.URL, strings.ReplaceAll(string(a.Content.ID), ":", "-"))) {
```

Comment:
It may be better to assign operations to each family.

Reply:
moved to vendor.go https://github.com/future-architect/vuls/pull/2106/commits/532998318445cf1326bad0befb6ab98b3191cd00#diff-1be336d4b7528308cbd3a4c85b9bdb0938f40b2c973f26011f7048bd7352a7cdR201